pkg/controllers/cspc-controller/scale.go

```diff
@@ -0,0 +1,184 @@
+/*
+Copyright 2020 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cspccontroller
+
+import (
+	"fmt"
+	"github.com/openebs/cstor-operators/pkg/cspc/algorithm"
+	"k8s.io/klog"
+	"github.com/openebs/api/pkg/apis/types"
+	cstor "github.com/openebs/api/pkg/apis/cstor/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/runtime"
+
+	"github.com/pkg/errors"
+
+)
+
+// ScaleUp creates as many cstor pool on a node as pendingPoolCount.
+func (c *Controller) ScaleUp(cspc *cstor.CStorPoolCluster, pendingPoolCount int) error {
+	for poolCount := 1; poolCount <= pendingPoolCount; poolCount++ {
+		err := c.CreateCSPI(cspc)
+		if err != nil {
+			message := fmt.Sprintf("Pool provisioning failed for %d/%d ", poolCount, pendingPoolCount)
+			c.recorder.Event(cspc, corev1.EventTypeWarning, "Create", message)
+			runtime.HandleError(errors.Wrapf(err, "Pool provisioning failed for %d/%d for cstorpoolcluster %s", poolCount, pendingPoolCount, cspc.Name))
```

Comment:
I think we have to revisit whether `runtime.HandleError()` is required or not since when an event is generated at the time itself log also will be generated.

Reply:
todo : revisit